### PR TITLE
Fix Amazon Providers mypy errors 

### PR DESCRIPTION
**Pull request by sunank200**
- Fix mypy errors for amazon hooks, sensors, triggers and example DAG.
 - Fix the test as per mypy changes

Part of issue: #65

### CODE REVIEW

File: astronomer/providers/amazon/aws/hooks/s3.py [hunk 1]
```diff
@@ -155,7 +153,7 @@ async def is_keys_unchanged(
         inactivity_seconds: int = 0,
         allow_delete: bool = True,
         last_activity_time: Optional[datetime] = None,
-    ) -> bool:
+    ) -> Dict[str, Any]:
```
Comment (kaxil) [inline astronomer/providers/amazon/aws/hooks/s3.py:156]:
```suggestion
    ) -> Dict[str, str]:
```

File: tests/amazon/aws/hooks/test_redshift_sql.py [hunk 2]
```diff
@@ -238,13 +238,8 @@ def test_execute_query_exception(mock_conn, mock_params):
...
     )
-    resp = hook.execute_query(TEST_SQL, params=None)
-    print(resp)
-    assert resp == {
-        "status": "error",
-        "message": "An error occurred (SomeServiceException) when calling the "
-        "redshift-data operation: Details/context around the exception or error",
-    }
+    with pytest.raises(AirflowException):
+        hook.execute_query(TEST_SQL, params=None)
```
Comment (kaxil) [inline tests/amazon/aws/hooks/test_redshift_sql.py:242]: you can still assert the exception message - https://stackoverflow.com/a/23514853/5691525

File: astronomer/providers/amazon/aws/hooks/redshift_data.py [hunk 3]
```diff
@@ -64,13 +67,17 @@ def get_conn_params(self) -> Dict[str, Union[str, int]]:
 
         return conn_params
 
-    def execute_query(self, sql: Union[str, Iterable[str]], params: Optional[Dict]):
+    def execute_query(
+        self, sql: Optional[Union[Dict[Any, Any], Iterable[Any]]], params: Optional[Dict[Any, Any]]
```
[the remark this answers is not in the document]
Comment (sunank200) [reply astronomer/providers/amazon/aws/hooks/redshift_data.py:71]: In [RedshiftSQLOperator](https://github.com/apache/airflow/blob/main/airflow/providers/amazon/aws/operators/redshift_sql.py#L56) the `sql: Optional[Union[Dict[Any, Any], Iterable[Any]]]` is Optional in constructor. If we remove it as optional, it will throw error as its inherited from Parent `RedshiftSQLOperator`.

[the remark this answers is not in the document]
Comment (kaxil) [reply astronomer/providers/amazon/aws/hooks/redshift_data.py:71]: I don't see it as `None`.

https://github.com/apache/airflow/blob/c7286e53064d717c97807f7ccd6cad515f88fe52/airflow/providers/amazon/aws/operators/redshift_sql.py#L56

[the remark this answers is not in the document]
Comment (kaxil) [reply astronomer/providers/amazon/aws/hooks/redshift_data.py:71]: aah looks like I fixed it: https://github.com/apache/airflow/pull/21885 🤦

File: astronomer/providers/amazon/aws/example_dags/example_redshift_cluster_management.py [hunk 4]
```diff
@@ -1,8 +1,8 @@
 import os
+from datetime import datetime
```
Comment (kaxil) [inline astronomer/providers/amazon/aws/example_dags/example_redshift_cluster_management.py:2]: Is this change required?

File: astronomer/providers/amazon/aws/operators/redshift_sql.py [hunk 5]
```diff
@@ -19,12 +19,12 @@ def __init__(
...
-        **kwargs,
+        **kwargs: Any,
     ) -> None:
         self.poll_interval = poll_interval
         super().__init__(**kwargs)
 
-    def execute(self, context: "Context"):
+    def execute(self, context: "Context") -> None:
         redshift_data_hook = RedshiftDataHook(aws_conn_id=self.redshift_conn_id)
         query_ids = redshift_data_hook.execute_query(sql=self.sql, params=self.params)
```
Comment (kaxil) [inline astronomer/providers/amazon/aws/operators/redshift_sql.py:29]:
```suggestion
        query_ids = redshift_data_hook.execute_query(cast(str, sql=self.sql), params=self.params)
```

File: astronomer/providers/amazon/aws/operators/redshift_sql.py [hunk 6]
```diff
@@ -1,6 +1,6 @@
-from typing import TYPE_CHECKING
+from typing import TYPE_CHECKING, Any, Dict
```
Comment (kaxil) [inline astronomer/providers/amazon/aws/operators/redshift_sql.py:1]:
```suggestion
from typing import TYPE_CHECKING, Any, Dict, cast
```

File: astronomer/providers/amazon/aws/hooks/redshift_data.py (hunk 3 above)
[the remark this answers is not in the document]
Comment (kaxil) [reply astronomer/providers/amazon/aws/hooks/redshift_data.py:71]: Following changes should take care of it:

- https://github.com/astronomer/astronomer-providers/pull/103/files#r821725255
- https://github.com/astronomer/astronomer-providers/pull/103/files#r821724962

File: astronomer/providers/amazon/aws/example_dags/example_redshift_cluster_management.py (hunk 4 above)
Comment (sunank200) [reply astronomer/providers/amazon/aws/example_dags/example_redshift_cluster_management.py:2]: mypy is throwing an error is we use airflow path for datatime

File: astronomer/providers/amazon/aws/hooks/redshift_data.py [hunk 7]
```diff
@@ -91,4 +98,4 @@ def execute_query(self, sql: Union[str, Iterable[str]], params: Optional[Dict]):
                 query_ids.append(response["Id"])
             return query_ids
         except botocore.exceptions.ClientError as error:
-            return {"status": "error", "message": str(error)}
+            raise AirflowException(str(error))
```
[the remark this answers is not in the document]
Comment (sunank200) [reply astronomer/providers/amazon/aws/hooks/redshift_data.py]: [Done](https://github.com/astronomer/astronomer-providers/pull/103/commits/6d97b057c9eff2c515860e896e73bf262a58431b)

File: astronomer/providers/amazon/aws/hooks/redshift_data.py (hunk 3 above)
[the remark this answers is not in the document]
Comment (kaxil) [reply astronomer/providers/amazon/aws/hooks/redshift_data.py:71]: Since we now use `cast` we should be good to remove `Optional`
